### ISSUE TITLE
Angular 10 compatibility issue

### ISSUE THREAD

**Issue by kcir3v4M** [rednez/angular-user-idle#96]
**Problem with a missing argument**
After upgrade to latest version of Angular (node_modules cleanup, package.json update, install), I cannot run my application because the following module is on error:

`ERROR in node_modules/angular-user-idle/lib/angular-user-idle.module.d.ts:4:45 - error TS2314: Generic type 'ModuleWithProviders<T>' requires 1 type argument(s).

4     static forRoot(config: UserIdleConfig): ModuleWithProviders;`

**To Reproduce**
1. Cleanup "node_modules"
2. Ensure that you are using the latest angular-cli and core
    `ng update @angular/cli @angular/core`
3. Update all dependencies
4. npm install
5. npm start

**Expected behavior**
No error in the console when I launch my web application

**Desktop**
 - OS: Windows 10
 - Browser Chrome (up-to-date)

`
  "dependencies": {
    "@angular/animations": "^10.1.4",
    "@angular/cdk": "^10.2.3",
    "@angular/common": "^10.1.4",
    "@angular/compiler": "^10.1.4",
    "@angular/core": "^10.1.4",
    "@angular/flex-layout": "^10.0.0-beta.32",
    "@angular/forms": "^10.1.4",
    "@angular/http": "^7.2.16",
    "@angular/material": "^10.2.3",
    "@angular/platform-browser": "^10.1.4",
    "@angular/platform-browser-dynamic": "^10.1.4",
    "@angular/router": "^10.1.4",
    "@ngrx/core": "^1.2.0",
    "@ngrx/effects": "^10.0.0",
    "@ngrx/router-store": "^10.0.0",
    "@ngrx/store": "^10.0.0",
    "@ngrx/store-devtools": "^10.0.0",
    "angular-in-memory-web-api": "^0.11.0",
    "angular-user-idle": "^2.2.4",
    "body-parser": "^1.19.0",
    "classlist.js": "^1.1.20150312",
    "core-js": "^3.6.5",
    "express": "^4.17.1",
    "hammerjs": "^2.0.8",
    "material-design-icons": "^3.0.1",
    "ngx-mock": "^2.0.1",
    "object-hash": "^2.0.3",
    "only": "^0.0.2",
    "rxjs": "^6.6.3",
    "rxjs-compat": "^6.6.3",
    "tslib": "^1.13.0",
    "uuid": "^8.3.1",
    "web-animations-js": "^2.3.2",
    "zone.js": "^0.11.1"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^0.1001.4",
    "@angular/cli": "^10.1.4",
    "@angular/compiler-cli": "^10.1.4",
    "@angular/language-service": "^10.1.4",
    "@types/jasmine": "^3.5.14",
    "@types/jasminewd2": "^2.0.8",
    "@types/node": "^14.11.2",
    "codelyzer": "^6.0.1",
    "faker": "^5.1.0",
    "jasmine": "^3.6.1",
    "jasmine-core": "^3.6.0",
    "jasmine-spec-reporter": "^6.0.0",
    "karma": "^5.2.3",
    "karma-chrome-launcher": "^3.1.0",
    "karma-coverage-istanbul-reporter": "^3.0.3",
    "karma-jasmine": "^4.0.1",
    "karma-jasmine-html-reporter": "^1.5.4",
    "node-sass": "^4.14.1",
    "node-windows": "^1.0.0-beta.5",
    "protractor": "^7.0.0",
    "ts-node": "^9.0.0",
    "tslint": "^6.1.3",
    "tslint-angular": "^3.0.3",
    "typescript": "^4.0.3"
  }
`

**Comment by SathishRavichandran** [comment]
Yes i'm also Running into same issue it is not working in angular 10

**Comment by rednez** [comment]
What version of angular-user-idle?

**Comment by rednez** [comment]
Not reproduced. I try in one of my current project:
1.` rm --rf node_modules`
2.` npm install`
3.` ng serve` and `ng build --prod`

`ng serve` and `ng build` makes successful build without any errors and warnings related to angular-user-idle.

Dependencies:
    ```
"@angular/animations": "~10.1.5",
    "@angular/cdk": "~10.2.4",
    "@angular/common": "~10.1.5",
    "@angular/compiler": "~10.1.5",
    "@angular/core": "~10.1.5",
    "@angular/forms": "~10.1.5",
    "@angular/material": "^10.2.4",
    "@angular/material-moment-adapter": "^10.2.4",
    "@angular/platform-browser": "~10.1.5",
    "@angular/platform-browser-dynamic": "~10.1.5",
    "@angular/router": "~10.1.5",
    "angular-user-idle": "2.2.4",
```

**Comment by michal-husak** [comment]
I have the same problem even after `npm ci`.

If you look at this built file `node_modules\angular-user-idle\lib\angular-user-idle.module.d.ts`:
```typescript
import { ModuleWithProviders } from '@angular/core';
import { UserIdleConfig } from './angular-user-idle.config';
export declare class UserIdleModule {
    static forRoot(config: UserIdleConfig): ModuleWithProviders;
}
```
`ModuleWithProviders` is not generic.

But it is generic in your source code:
```typescript
@NgModule({
  imports: []
})
export class UserIdleModule {
  static forRoot(config: UserIdleConfig): ModuleWithProviders<UserIdleModule> {
    return {
      ngModule: UserIdleModule,
      providers: [
        {provide: UserIdleConfig, useValue: config}
      ]
    };
  }
}
```

Perhaps the library build is configured in a wrong way?

**Comment by michal-husak** [comment]
I just cloned your repo, built it and it is correct. But in my `node_modules` of the testing app it is still the old `angular-user-idle.module.d.ts`.

If you download the library npm package directly from npm registry it has still the old build. I assume it was not published to npm?

**Comment by toski112** [comment]
Any ideas to fix this problem?

**Comment by rednez** [comment]
I published 2.2.5 into npm. Please check it. I hope it works :)
